src/__tests__/request_middleware.test.ts

```diff
@@ -94,15 +172,32 @@ describe("withRequestMiddlewares", () => {
     });
   });
 
+  it("should return an IResponseErrorInternal in case middleware fails", () => {
+    const mockHandler = jest.fn(() => Promise.resolve(response));
+    const handler = withRequestMiddlewares(DelayedFailureMiddleware)(mockHandler);
+
+    return handler(request as any).then(r => {
+      expect(mockHandler).not.toHaveBeenCalled();
+      expect(r.kind).toBe("IResponseErrorInternal");
+    });
+  });
+
   it("should stop processing middlewares after a rejecting middleware", () => {
```

Comment:
Ca we add a test to ensure that in a pipeline of N middlewares (even with 2 or more RejectingMiddleware) we get only the first rejecting middleware output?

Reply:
Like every other middleware defined after a rejecting one, the second rejecting middleware will not be executed, so its output can not be included in output.
I improved the existing test in [1600bd0](https://github.com/pagopa/io-ts-commons/pull/370/commits/1600bd0a48f283692b0b08cddc094888413d871f) to make it explicit